src/main/java/net/dries007/tfc/objects/te/TECharcoalForge.java

```diff
@@ -87,6 +87,15 @@ public void update()
             IBlockState state = world.getBlockState(pos);
             if (state.getValue(LIT))
             {
+                // Have to check the above block, since minecraft think this block is "roof"
+                if (world.isRainingAt(pos.up()))
+                {
+                    // Instantly consume last fuel and turn off
+                    burnTicks = 0;
+                    burnTemperature = 0;
+                    world.setBlockState(pos, state.withProperty(LIT, false));
+                    return;
+                }
```

Comment:
Is there a particular reason this both 
 - checks every tick for rain instead of on random ticks or % chances, and
 - has a 100% chance to go out on said exact tick?

Given the charcoal forge has a requirement that it must see sky light (although it can be indirect), I would assume a much lower chance of going out in the rain. It's also a charcoal bleedin forge that can reach temperatures > 1500 C, but rain puts it out *instantly*?